h2o-algos/src/main/java/hex/tree/gbm/GBM.java

```diff
@@ -182,6 +179,9 @@
     if ((_train != null) && (_parms._monotone_constraints != null)) {
       TreeUtils.checkMonotoneConstraints(this, _train, _parms._monotone_constraints);
     }
+    if (!Arrays.stream(supportedCategoricalEncodingSchemes()).anyMatch(_parms._categorical_encoding::equals)) {
+      error("_categorical_encoding", _parms._categorical_encoding + " is not supported for GBM.");
```

Comment:
1. Do we want to compare ignoring case ? Users can always make a mistake in lower/upper case. I believe we're benevolent to this kind of user-inpuit mistakes in the API cc @michalkurka .
2. The `contains; method of a set would be faster and would not require allocating additional memory by calling `Arrays.stream`, but this is just something to think about. These arrays are small and the change won't be noticeable. But it's another piece of evidence to use Set, if possible due to serialization reasons.
3. Would be nice if we printed the list of available encodings right away.